src/libraries/TickMath.sol

```diff
@@ -49,8 +49,13 @@ library TickMath {
             uint256 absTick = tick < 0 ? uint256(-int256(tick)) : uint256(int256(tick));
             if (absTick > uint256(int256(MAX_TICK))) revert InvalidTick();
 
-            uint256 price =
-                absTick & 0x1 != 0 ? 0xfffcb933bd6fad37aa2d162d1a594001 : 0x100000000000000000000000000000000;
+            // Equivalent to:
+            //     price = absTick & 0x1 != 0 ? 0xfffcb933bd6fad37aa2d162d1a594001 : 0x100000000000000000000000000000000;
+            //     or price = int(2**128 / sqrt(1.0001)) if (absTick & 0x1) else 1 << 128
+            uint256 price;
+            assembly {
+                price := xor(shl(128, 1), mul(xor(shl(128, 1), 0xfffcb933bd6fad37aa2d162d1a594001), and(absTick, 0x1)))
```

Comment:
does the optimiser deal with cutting down this line? theres lots of things it knows it advance
`xor(shl(128, 1), 0xfffcb933bd6fad37aa2d162d1a594001)`
`shl(128, 1)`

Reply:
Yes. The compiler optimizer is able to reduce literal math expressions.